### PR TITLE
Update how AffixAllocator infers shared from parent

### PR DESCRIPTION
**Pull request by edi33416**
`AffixAllocator` used to only check if `Allocator.instance` is `shared`. If the parent `Allocator` doesn't have an `instance` field, the check will be false, even though the `Allocator` is shared.

### CODE REVIEW

File: std/experimental/allocator/building_blocks/affix_allocator.d [hunk 1]
```diff
@@ -499,3 +504,16 @@ struct AffixAllocator(Allocator, Prefix, Suffix = void)
     assert(b.length == 100);
     assert((() nothrow @nogc => a.deallocate(b))());
 }
+
+@system shared unittest
+{
+    import std.experimental.allocator : processAllocator, RCISharedAllocator;
+
+    alias SharedAllocT = shared AffixAllocator!(RCISharedAllocator, int);
+    assert(is(RCISharedAllocator == shared));
```
Comment (ZombineDev) [inline std/experimental/allocator/building_blocks/affix_allocator.d]: You should also add an assert which verifies that `SharedAllocT`'s methods are actually inferred as `shared`.

Comment (ZombineDev) [reply std/experimental/allocator/building_blocks/affix_allocator.d]: I'm not sure how annotating the unittest with `shared` helps.

**Comment by andralex** [comment]
@ZombineDev can you please add the assert?

**Comment by edi33416** [comment]
I updated the unittest and added `static assert`s;

**Pull request merged**